scripts/run_example

```diff
@@ -1,145 +0,0 @@
-
-
-if [[ "${server}" != "none" ]]; then
-  "${SCRIPTS_DIR}/build_example" ${buildargs} -a "${application_language}" -c "${client_variants}" -e "${EXAMPLE}"
-
-  # Run a server in the background.
-  # The server is being built before running, so the build process will not happen in the
-  # background.
-  "${SCRIPTS_DIR}/build_server" -s "${server}"
-  "${SCRIPTS_DIR}/run_server" ${serverargs} -s "${server}" -a "${application_language}" -e "${EXAMPLE}" &
-  readonly SERVER_PID=$!
-  to_kill+=("${SERVER_PID}")
-
-  # Run Backend for specific examples.
-  if [[ "${EXAMPLE}" == 'aggregator' ]] || [[ "${EXAMPLE}" == 'trusted_information_retrieval' ]]; then
-    # Building the Backend first, so it will not be built in the background.
-    cargo build --release --target="${RUST_HOST_TARGET}" --manifest-path="examples/${EXAMPLE}/backend/Cargo.toml"
-    cargo run --release --target="${RUST_HOST_TARGET}" --manifest-path="examples/${EXAMPLE}/backend/Cargo.toml" -- \
-      --grpc-tls-private-key="${SCRIPTS_DIR}/../examples/certs/local/local.key" \
-      --grpc-tls-certificate="${SCRIPTS_DIR}/../examples/certs/local/local.pem" &
-    readonly BACKEND_PID=$!
-    to_kill+=("${BACKEND_PID}")
-  fi
-
-  sleep 3  # Wait for the application to start.
-fi
-
-if [[ "${EXAMPLE}" == 'abitest' ]]; then
-  # TODO(#1040): reinstate storage tests when Rust runtime supports them.
-  readonly ADDITIONAL_ARGS=('--test_exclude=Storage')
-else
-  readonly ADDITIONAL_ARGS=()
-fi
-
-if [[ "${EXAMPLE}" == 'abitest' ]]; then
-  readonly TLS_ARGS=(
-    "--ca_cert=${SCRIPTS_DIR}/../examples/certs/local/ca.pem"
-    "--cert_chain=${SCRIPTS_DIR}/../examples/certs/local/local.pem"
-    "--private_key=${SCRIPTS_DIR}/../examples/certs/local/local.key")
-else
-  readonly TLS_ARGS=("--ca_cert=${SCRIPTS_DIR}/../examples/certs/local/ca.pem")
-fi
-
-readonly CLIENT_ARGS=("${@-}")  # Choose client args provided after '--'.
-
-# Run the application clients.
-for client_variant in ${client_variants}
-do
-  case "${client_variant}" in
-    cargo)
-      rust_client_manifest="${SCRIPTS_DIR}/../examples/${EXAMPLE}/client/rust/Cargo.toml"
-          cargo run --release --target="${RUST_HOST_TARGET}" --manifest-path="${rust_client_manifest}" -- \
-            --root-tls-certificate="${SCRIPTS_DIR}/../examples/certs/local/ca.pem" \
-            "${CLIENT_ARGS[@]-}"
-      ;;
-    bazel)
-          "${SCRIPTS_DIR}/../bazel-client-bin/examples/${EXAMPLE}/client/client" "${TLS_ARGS[@]}" "${ADDITIONAL_ARGS[@]-}" "${CLIENT_ARGS[@]-}"
-      ;;
-    npm)
-      npm_client="${SCRIPTS_DIR}/../examples/${EXAMPLE}/client/nodejs"
-          npm start --prefix "${npm_client}"
-      ;;
-    *)
-      echo "Invalid example client variant: ${client_variant}"
-      exit 1;;
-  esac
-done
-
-if [[ "${server}" != "none" ]]; then
-  # Check whether the server is still running.
-  ps -p ${SERVER_PID} > /dev/null
-  readonly SERVER_GONE=$?
-  if ${SERVER_GONE} ; then
-    echo "Runtime server has terminated early"
```

Comment:
On a quick play, I couldn't see an equivalent check to this in `runner`.

Reply:
Added a TODO, I'd like to find a nicer way for this, and I think the situation is somewhat mitigated in runner by the fact that if the background process exits with a nonzero status, it will already count as failure.